src/interface/src/app/services/plan.service.ts

```diff
@@ -71,12 +92,21 @@ export class PlanService {
     this.planState$.next(updatedState);
   }
 
-  /** Temporary stub for the api. */
-  private createPlanMockApi(plan: BasePlan): Plan {
-    this.tempPlanId = ++this.tempPlanId;
+  private createPlanApi(plan: BasePlan): Plan {
+    const createPlanRequest = this.convertToDbPlan(plan);
+    ++this.tempPlanId;
+    var planId: number = this.tempPlanId;
+    this.http
+      .post(BackendConstants.END_POINT + '/plan/create/', createPlanRequest, {
+        withCredentials: true,
+      })
+      .subscribe((result) => {
+        planId = Number(result.toString());
+      });
+
     return {
       ...plan,
-      id: this.tempPlanId.toString(),
+      id: String(planId),
     };
```

Comment:
I think this method needs to return an Observable. Currently, if I understand my rxjs, the method will return before the HTTP request returns a result, so planID won't be assigned in time. If this method returns an observable, then line 48 doesn't need to wrap the result in of() anymore.

Reply:
Fixed.